pkg/addonmanager/addonclients/fluxaddonclient.go

```diff
@@ -52,11 +52,35 @@ const (
 	deleteClusterconfigCommitMessage  = "Delete commit of cluster configuration; generated by EKS-A CLI"
 )
 
+// GitClient is responsible for provider-agnostic actions on a generic git repository.
```

Comment:
I think, with this comment, we could make it clearer that the backing construct represents a _single_ repository.